Code/Framework/AzCore/AzCore/Memory/SimpleSchemaAllocator.h

```diff
@@ -19,19 +19,21 @@ namespace AZ
     /**
     * A basic, default allocator implementation using a custom schema.
     */
-    template <class Schema, class DescriptorType=typename Schema::Descriptor, bool ProfileAllocations=true, bool ReportOutOfMemory=true>
+    template <class Schema, bool ProfileAllocations=true, bool ReportOutOfMemory=true>
     class SimpleSchemaAllocator
         : public AllocatorBase
     {
     public:
-        using Descriptor = DescriptorType;
-        using pointer_type = typename Schema::pointer_type;
+        AZ_RTTI(((SimpleSchemaAllocator<Schema>), "{32019C72-6E33-4EF9-8ABA-748055D94EB2}", Schema), AllocatorBase)
```

Comment:
There inner set of parentheses isn't needed here because `SimpleSchemaAllocator<Schema>` doesn't contain a comma.
This also messes up the type name when queried using `AZ::TypeInfo<T>::Name()`.

This would output the name as "(SimpleSchemaAllocator\<Schema>)<_ActualSchemaTemplateParameter>_".

The AZ_TYPE_INFO macros automatically appends the template argument to the "_ClassName" paramtter and therefore only "SimpleSchemaAllocator" should be specified.
```suggestion
        AZ_RTTI((SimpleSchemaAllocator, "{32019C72-6E33-4EF9-8ABA-748055D94EB2}", Schema), AllocatorBase)
```

Reply:
Fixed.